sdks/go/pkg/beam/core/graph/coder/varint.go

```diff
@@ -64,12 +78,39 @@ func decVarIntZ(t reflect.Type, data []byte) (typex.T, error) {
 	if size <= 0 {
 		return nil, fmt.Errorf("invalid varintz encoding for: %v", data)
 	}
-	return reflect.ValueOf(n).Convert(t).Interface(), nil
+	switch t.Kind() {
+	case reflect.Int:
+		return int(n), nil
+	case reflect.Int8:
+		return int8(n), nil
+	case reflect.Int16:
+		return int16(n), nil
+	case reflect.Int32:
+		return int32(n), nil
+	case reflect.Int64:
+		return n, nil
+	}
+	panic(fmt.Sprintf("unreachable statement: expected a signed integer, got %v", t))
```

Comment:
nit: move the panic into a default branch. That way a forgotten "return" is a compile-time error.